batch/test/test_batch.py

```diff
@@ -674,6 +677,103 @@ def test_verify_no_access_to_metadata_server(client):
     assert "Connection timed out" in j.log()['main'], (str(j.log()['main']), status)
 
 
+def test_submit_batch_in_job(client):
+    builder = client.create_batch()
+    bucket_name = get_user_config().get('batch', 'bucket')
+    script = f'''import hailtop.batch as hb
+backend = hb.ServiceBackend("test", "{bucket_name}")
+b = hb.Batch(backend=backend)
+j = b.new_bash_job()
+j.command("echo hi")
+b.run()
+backend.close()
+'''
+    j = builder.create_job(
+        os.environ['HAIL_HAIL_BASE_IMAGE'],
+        ['/bin/bash', '-c', f'''python3 -c \'{script}\''''],
+        mount_tokens=True,
+    )
+    builder.submit()
+    status = j.wait()
+    assert status['state'] == 'Success', str(status)
+
+
+def test_cant_submit_to_default_with_other_ns_creds(client):
+    bucket_name = get_user_config().get('batch', 'bucket')
+    script = f'''import hailtop.batch as hb
+backend = hb.ServiceBackend("test", "{bucket_name}")
+b = hb.Batch(backend=backend)
+j = b.new_bash_job()
+j.command("echo hi")
+b.run()
+backend.close()
+'''
+
+    builder = client.create_batch()
+    j = builder.create_job(
+        os.environ['HAIL_HAIL_BASE_IMAGE'],
+        [
+            '/bin/bash',
+            '-c',
+            f'''
+rm /deploy-config/deploy-config.json
+python3 -c \'{script}\'''',
+        ],
+        mount_tokens=True,
+    )
+    builder.submit()
+    status = j.wait()
+    if NAMESPACE == 'default':
+        assert status['state'] == 'Success', str(status)
+    else:
+        assert status['state'] == 'Failed', str(status)
+
+    builder = client.create_batch()
+    j = builder.create_job(
+        os.environ['HAIL_HAIL_BASE_IMAGE'],
+        [
+            '/bin/bash',
+            '-c',
+            f'''
+jq '.domain = "batch.{DOMAIN}"' /deploy-config/deploy-config.json > /deploy-config/deploy-config.json
+python3 -c \'{script}\'''',
+        ],
+        mount_tokens=True,
+    )
+    builder.submit()
+    status = j.wait()
+    if NAMESPACE == 'default':
+        assert status['state'] == 'Success', str(status)
+    else:
+        assert status['state'] == 'Failed', str(status)
+
+
+def test_cannot_contact_other_internal_ips(client):
+    internal_gateway_last_byte = int(INTERNAL_GATEWAY_IP.rsplit('.', maxsplit=1)[1])
+    assert internal_gateway_last_byte <= 253
+    other_ip = f'10.128.0.{internal_gateway_last_byte + 1}'
+    other_ip_2 = f'10.128.0.{internal_gateway_last_byte + 2}'
+    builder = client.create_batch()
+    script = f'''
+if [ -z ${{HAIL_BATCH_WORKER_IP+x}} ]; then
```

Comment:
It could be that the current worker is neither of the given IPs, which is also fine, I just don't want to accidentally collide. I My checking to make sure it is set is a little defensive, since using `port=5000` should guarantee that it is. If someone were to remove `port=5000`, then that environment variable wouldn't be set and we wouldn't know the batch worker's IP, at which point we can't guarantee that the job won't accidentally try to ping its own worker.

Reply:
I thought about this a bit and I think that check just made the test harder to understand at a glance because of the guarantees we have around that environment variable being set. I've removed that check.